packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -445,6 +415,93 @@ export class Pipeline extends PipelineBase {
     return actionRole;
   }
 
+  private getRoleFromActionPropsOrGenerateIfCrossAccount(stage: Stage, action: IAction): iam.IRole | undefined {
+    if (action.actionProperties.role) {
+      if (this.isAwsOwned(action)) {
+        return action.actionProperties.role;
+      } else {
+        throw new Error("Specifying a Role is not supported for actions with an owner different than 'AWS' - " +
```

Comment:
The terminology "owner" here is extremely confusing (especially since we use it to indicate "owned resources"). How about something like "The action X does not support specifying a role".

Reply:
Just to be clear: I'm talking here about [this property](https://github.com/aws/aws-cdk/blob/cd0e9bd53e37f4b5824d64a0234e28c3a86b1ed9/packages/%40aws-cdk/aws-codepipeline/lib/action.ts#L63).

Given that, does the error message makes sense? (I'm not worried about the `isAwsOwned()` method, it's private anyway)